include/Link.hpp

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include "NetworkDevice.hpp"
+
+#include <cstdint>
+#include <memory>
+
+class Link {
+public:
+    Link(std::shared_ptr<NetworkDevice> source, std::shared_ptr<NetworkDevice> destination, std::uint64_t distance_ns);
```

Comment:
I'd suggest to use `latency_ns` instead of `distance_ns`.

The term "latency" specifically refers to the time delay experienced in a system, particularly in the context of data transmission. By using `distance_ns` you misleadingly assign time units to the distance which is not very technical.

```suggestion
    Link(std::shared_ptr<NetworkDevice> source, std::shared_ptr<NetworkDevice> destination, std::uint64_t latency_ns);
```

Reply:
Thanks. I do not see a big problem with `distance_ns` as we do measure distances in the light years in astronomy, but I would suggest `delay_ns`.